spec/services/badges/award_contributor_from_github_spec.rb

```diff
@@ -6,9 +6,23 @@
   before do
     badge
     omniauth_mock_github_payload
+    allow(Settings::Authentication).to receive(:providers).and_return([:github])
     stub_const("#{described_class}::REPOSITORIES", ["forem/DEV-Android"])
   end
 
+  it "won't work without Github oauth configured" do
+    allow(Settings::Authentication).to receive(:providers).and_return([])
+    user = create(:user, :with_identity, identities: ["github"], uid: "389169")
+
+    Timecop.freeze("2021-08-16T13:49:20Z") do
+      expect do
+        VCR.use_cassette("github_client_commits_contributor_badge") do
+          described_class.call
+        end
+      end.to change(user.badge_achievements, :count).by(0)
+    end
```

Comment:
Do we need the request stubbing and Timecop and all that? It shouldn't be attempting to make the API call at all, right?

```suggestion
    expect { described_class.call }
      .not_to change { user.badge_achievements.count }
```

Also, I tweaked the formatting of this suggestion because I think it reads a whole lot more clearly. I don't know why we use the other form, tbh. I find it so much harder to understand.

Reply:
You are right :smile: I can definitely simplify this whole thing.